topo/topo.go

```diff
@@ -598,23 +614,55 @@ func (m *Manager) push(ctx context.Context) error {
 	}
 
 	log.Infof("Creating Node Pods")
+
 	for _, n := range m.nodes {
-		for key, service := range n.GetProto().Services {
-			updateServicePortName(service, key)
-		}
-		if err := n.Create(ctx); err != nil {
-			return fmt.Errorf("failed to create node %s: %w", n, err)
-		}
-		log.Infof("Node %s resource created", n)
+		wg.Add(1)
+		go func(nodeToCreate node.Node) {
+			defer wg.Done()
+
+			for key, service := range nodeToCreate.GetProto().Services {
+				updateServicePortName(service, key)
+			}
+
+			if err := nodeToCreate.Create(ctx); err != nil {
+				errCh <- fmt.Errorf("failed to create node %s: %w", nodeToCreate, err)
+				return
+			}
+			log.Infof("Node %s resource created", nodeToCreate)
+		}(n)
 	}
+	wg.Wait()
+
+	select {
+	case err := <-errCh:
+		return err
+	default:
+	}
+
+	log.Infof("Generating Self-Signed Certificates")
```

Comment:
why not move the self signed gen as part of the previous func then you don't have two fan out / fan in